src/snowflake/snowpark/functions.py

```diff
@@ -2479,13 +2832,24 @@ def log(
         >>> df.select(log(10, df["a"]).cast(IntegerType()).alias("log")).collect()
         [Row(LOG=0), Row(LOG=1)]
     """
-    b = lit(base) if isinstance(base, (int, float)) else _to_col_if_str(base, "log")
-    arg = lit(x) if isinstance(x, (int, float)) else _to_col_if_str(x, "log")
-    return builtin("log")(b, arg)
+    b = (
+        lit(base, _emit_ast=_emit_ast)
+        if isinstance(base, (int, float))
+        else _to_col_if_str(base, "log")
+    )
+    arg = (
+        lit(x, _emit_ast=_emit_ast)
+        if isinstance(x, (int, float))
+        else _to_col_if_str(x, "log")
+    )
```

Comment:
shouldn't `_emit_ast=False` in all these cases?

Reply:
yup, we can do that. It actually doesn't matter here because `lit(..)` builds an expression but not a statement. Only statements get flushed to the _ast_batch, expressions live in main memory (until used in a stmt). Leaving _emit_ast here with pass-through is preferential because the builtin will then copy out simply the ast of the lit arguments. If False, if _emit_ast is True the builtin will create expressions from the lit anyways.

Reply:
resolved offline